tests/foreman/endtoend/utils.py

```diff
@@ -21,6 +22,8 @@ def client_provisioning(
             is available.
         :param package_name: Name of the package to be installed on the client.
         """
+        if not setting_is_set('clients'):
```

Comment:
skip_if_not_set is not a function but a decorator. It have the same functionality of is_bz_bug_open for example, it is a function that allows skip_if_not_set do its job and it is also made available public to that module for usage outside the module.